### ISSUE TITLE
Consumers start hanging with AIOKafkaConnection._read() Cancelled Error

### ISSUE THREAD

**Issue by vineet-rh** [aio-libs/aiokafka#252]
The consumers start handling at some point with the following error being logged:

```
[2017-11-07 18:17:47,547: ERROR] Task exception was never retrieved
future: <Task finished coro=<AIOKafkaConnection._read() done, defined at /home/robinhood/env/lib/python3.6/site-packages/aiokafka/conn.py:194> exception=CancelledError()>
```

What might be causing the error? Should we be crashing the consumer upon getting the error?

**Comment by vineet-rh** [comment]
I confirmed that this isn't making the consumer hang and is not really having any adverse effects. However, we should be handling these exceptions better.

**Comment by vineet-rh** [comment]
Any updates on this?

I repeatedly keep getting these errors, don't think it is fatal though.

**Comment by tvoinarovskyi** [comment]
Could you describe how you use the consumer? The error seems strange, it's like you close the consumer the same moment it connects... It is probably not critical, but I still fail to understand when it fails.

**Comment by vineet-rh** [comment]
I am not closing the consumer. This happens while the consumer is idle for sometime waiting for messages on certain partitions.

**Comment by tvoinarovskyi** [comment]
Ohh, this is probably a fail on idle connection closure. The connection is closed if not active for some time, probably this log is a side effect there. Will investigate, thank

**Comment by vsel** [comment]
I have the same problem. After 10 minutes of idle connection.

**Comment by tvoinarovskyi** [comment]
Ignore it, it's a bug, but should have no impact on operability

**Comment by vsel** [comment]
OK. Thank you. I was confused because I just produce data and read nothing.

**Comment by tvoinarovskyi** [comment]
@vsel Sorry, but I still have problems reproducing this. Could you provide your Producer configuration and if possible the script, that can reproduce the issue.

**Comment by tvoinarovskyi** [comment]
Ok, this is not the `idle` thing... The `connections_max_idle_ms` config did not work at all =)

**Comment by vsel** [comment]
Indeed! Do you still need my configuration for reproducing the issue? P.S. Sorry for delay.

**Comment by tvoinarovskyi** [comment]
Yes, please. Want to track it down

**Comment by vsel** [comment]
I can’t reproduce this bug with virtual kafka environment, so I’ll try to chat about this issue with devOps colleagues. I’m sorry. May be I can check something on production environment?

**Comment by tvoinarovskyi** [comment]
This should be fixed as of 0.4.0, but will leave open until someone can confirm it.

**Comment by bt-wil** [comment]
I haven't ran into this issue using `aiokafka==0.4.0`

**Comment by myaspm** [comment]
I'm still getting this errors with version 0.4.0. Don't think that it impacts anything but still..
